specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json

```diff
@@ -194,6 +194,39 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/availabilitySets": {
+      "get": {
+        "tags": [
+          "AvailabilitySets"
+        ],
+        "operationId": "AvailabilitySets_ListAll",
+        "description": "Gets the list of availability sets in the subscription. Use nextLink property in the response to get the next page of availability sets. Do this till nextLink is null to fetch all the availability sets.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/AvailabilitySetListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
```

Comment:
@gatneil `AvailabilitySetListResult` doesn't have `nextLink`. According to other usages of `AvailabilitySetListResult`, the `nextLinkName` should be `null`, as described [here](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable).

Reply:
But this API does have a nextLink; the example (which is a scrubbed version of a response from the live rest api) shows it: https://github.com/Azure/azure-rest-api-specs/pull/1584/files#diff-cd68bd619348d807b165aaa12af94817R662

Reply:
If API has the `nextLink` property then the response schema (in this case `AvailabilitySetListResult`) should have it as well. For example https://github.com/gatneil/azure-rest-api-specs/blob/b559e04a1c2608884f10d29c8be90df5fe75b464/specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json#L3674. To avoid breaking changes, I would suggest add `nextLink` as a read only property to `AvailabilitySetListResult`.

Reply:
BTW, In this example https://github.com/gatneil/azure-rest-api-specs/blob/b559e04a1c2608884f10d29c8be90df5fe75b464/specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json#L162, the response schema is also `AvailabilitySetListResult` but the "nextLinkName" is `null`.

Reply:
I see...I'll investigate a bit then re-visit this PR :)

Reply:
@gatneil thank you

Reply:
@gatneil let me know if you need more information or would like to discuss this offline/skype.